### PR TITLE
Remove special treatment of boundary cells

### LINKED ISSUE #12665: fullydistributed triangulation made from description incl. manifolds results in disjoint faces across at subdomain borders
Hi everyone,
I might have found a bug in parallel::fullydistributed::Triangulation (pfdt) when I was trying to create a pfdt from a serial triangulation, although it might also be rooted in wrong use of the functions.
My testcase is a circular disk in 2D with a circular inclusion which I read from an .msh file (a similar mesh can be created combining a `hyper_shell` and a `hyper_ball`):
circ_inclusion
Then I do the following:
1) Read in a grid with `read_msh()` into a serial triangulation.
2) In cells touching the boundary `cell->set_all_manifold_ids(100)`
3) Attach a `SphericalManifold` to both the serial triangulation and pfdt (similar as in `tests/fullydistributed_grids/copy_serial_tria_03.cc`)
4) Refine serial triangulation via `refine_global()`
5) Partition serial triangulation via `partition_triangulation`
6) Create description of the serial triangulation via `create_description_from_triangulation`
7) Actually create pfdt via `create_triangulation()`
If I do so, the created pfdt features disjoint cell faces across subdomains, as can be seen here:
circ_inclusion_3_procs
I can also replace 6) and 7) by `copy_triangulation()`, which gives the expected results, so, no disjoint cells, but also has a completely different partitioning.
But am I using above mentioned function(s) in a wrong way?
Any comments, suggestions, etc. are highly welcome!
My MWE plus the mesh file (in dummy .txt file format):
[circ_inclusion.txt](https://github.com/dealii/dealii/files/7006226/circ_inclusion.txt)
[copy_serial_tria_03.txt](https://github.com/dealii/dealii/files/7006227/copy_serial_tria_03.txt)

EDIT:
If one does not attach a manifold at all, 6) and 7) do work as expected and in `tests/fullydistributed_grids/copy_serial_tria_03.cc` all cells do have the same manifold attached to them. So it seems, that a cell does not know, which manifolds are attached on other processors to map the mesh in a conforming way (its assuming FlatManifold?) - which sounds reasonable! I guess this functionality is simply impossible due to the design of pfdt?

### PR DESCRIPTION
**Pull request by peterrum**
alternative to #12669, see also comment https://github.com/dealii/dealii/issues/12665#issuecomment-901323070

closes #12665

Let's see what happens.

### CODE REVIEW

**Review by kronbichler — COMMENTED** [review]
I think this is the way to go. I ran the testsuite on this change already and you need to adjust the following test outputs:
- `manifold/polar_manifold_02`
- `manifold/manifold_id_06.debug`
- `fullydistributed_grids/repartitioning_05.mpirun=4`

I am not entirely sure about the failures because we have an annoyingly long list of intermittent failures, see e.g. https://cdash.43-1.org/viewTest.php?onlyfailed&buildid=4535 , but I hope that this is it. I will check the grid in some of the examples to see if I am happy with the output as well.

**Comment by kronbichler** [comment]
I looked at the grids generated by `manifold/polar_manifold_02` and they do look slightly different than before, but neither is better. That case is a bit strange because we have a polar manifold attached to all cells of a disk (and a 3D ball!) in touch with the boundary, which leads to a sudden change from a flat manifold (inner cell) to curved manifolds. That grid would be better with a transfinite interpolation manifold, but that apart, this patch is surely a good simplification. The number of times we remove the old "boundary" concept is worth it :wink:

**Review by kronbichler — APPROVED** [review]
Thanks, this looks great!

**Comment by richardschu** [comment]
Just rerun the tests and from what i can see this works like a charm!
Thanks again! :+1:

**Review by drwells — APPROVED** [review]

File: source/grid/tria.cc [hunk 1]
```diff
@@ -3737,49 +3737,11 @@ namespace internal
...
-
-                // determine middle vertex by transfinite interpolation to be
-                // consistent with what happens to quads in a Triangulation<3,
-                // 3> when they are refined
-                triangulation.vertices[next_unused_vertex] =
-                  cell->center(true, true);
-              }
+            // determine middle vertex by transfinite interpolation to be
+            // consistent with what happens to quads in a Triangulation<3,
+            // 3> when they are refined
```
Comment (bangerth) [inline source/grid/tria.cc:3742]: Can you try to wrap the template arguments more nicely?

**Review by bangerth — COMMENTED** [review]
In the previous code, we cleared the user flag. Is this no longer necessary? Are we even still using the user flags now?

**Comment by peterrum** [comment]
@bangerth I have updated the comment. Regarding:

> In the previous code, we cleared the user flag. Is this no longer necessary? Are we even still using the user flags now?

They are not cleared anymore, since they are not set anymore.

**Pull request merged**